### ISSUE TITLE
Ts generation with known types

### ISSUE THREAD

**Issue by pirobox** [RicoSuter/NJsonSchema#1189]
Good afternoon,
   I can't find a way for solving this situation.
I'm trying to generate typescript interfaces from c# types.
C# types are something like this
class Person
{
   public string Name
   public string Address
}
class Address
{
   // Not important
}

NOW, my difficulty is that in my very old software, the ts interface for Address is already defined in my code base and i'd like to reuse it.
In particular my ts interface for address is defined in a namespace.
namespace NS1
{
   interface Address{}
}

So, basically, I would like to autogenerate a ts interface Person that is something like this
interface Person
{
   Address: NS1.Address
}

Is this possible?

Thx in advance

**Comment by RicoSuter** [comment]
Isnt there an ExcludedTyoes property on thr ts gen settings? If so it would however generate without the NS... another option is to write an own cli snd use the njs pkgs - this way you can get a collection of code artifacts instead of a single code output and exclude the unwanted type before rendering

**Comment by pirobox** [comment]
I'm actually integrating the njs packages in a custom cli.
May you elaborate a bit more your answer about generating a collection of code artifacts and exclude unwanted types before rendering?

**Comment by RicoSuter** [comment]
You can generate artifacts (not a single file) and then you have the code for each class/interface/etc. and you can filter on that before generating a single file. 

https://github.com/RicoSuter/NJsonSchema/blob/master/src/NJsonSchema.CodeGeneration.TypeScript/TypeScriptGenerator.cs#L54
